src/go/k8s/pkg/resources/certmanager/kafka_api.go

```diff
@@ -67,22 +67,24 @@ func (r *PkiReconciler) prepareKafkaAPI(
 	nodeSecretRef := tlsListener.NodeSecretRef
 
 	if nodeSecretRef == nil {
-		// Redpanda cluster certificate for Kafka API - to be provided to each broker
-		cn := NewCommonName(r.pandaCluster.Name, RedpandaNodeCert)
-		certsKey := types.NamespacedName{Name: string(cn), Namespace: r.pandaCluster.Namespace}
+		certName := NewCertName(r.pandaCluster.Name, RedpandaNodeCert)
+		certsKey := types.NamespacedName{Name: string(certName), Namespace: r.pandaCluster.Namespace}
 		nodeIssuerRef := issuerRef
 		if externalIssuerRef != nil {
 			// if external issuer is provided, we will use it to generate node certificates
 			nodeIssuerRef = externalIssuerRef
 		}
 
-		dnsName := []string{r.internalFQDN}
 		externalListener := r.pandaCluster.ExternalListener()
-		if externalListener != nil && externalListener.External.Subdomain != "" {
-			dnsName = append(dnsName, externalListener.External.Subdomain)
+		internalListener := r.pandaCluster.InternalListener()
+		dnsNames := []string{}
+		if internalListener != nil && internalListener.TLS.Enabled {
+			dnsNames = append(dnsNames, r.internalFQDN)
 		}
-
-		redpandaCert := NewNodeCertificate(r.Client, r.scheme, r.pandaCluster, certsKey, nodeIssuerRef, dnsName, cn, false, keystoreSecret, r.logger)
+		if externalListener != nil && externalListener.TLS.Enabled && externalListener.External.Subdomain != "" {
```

Comment:
so if this is external without subdomain, the dnsnames will be empty. I assume that's fine (?)

Reply:
🤔 I guess the only case where this would happen is if we want certs for the IPs of the nodes?
How can I get IPs of the nodes?

Reply:
Looks like currently we don't support certificates for IPs, so I just added a TODO.
If you think it can be quickly added please let me know how and I'll add it to the PR, if not, let's tackle it in a future PR.

Reply:
yeah I just wonder if the cert even will get created when dns names are empty (because before there were always at least the internal dns names). I don't think we should add the ips ... we discussed it, it's pretty complicated considering that in cloud environment nodes can come and go and you would have to reissue the certs

Reply:
> yeah I just wonder if the cert even will get created when dns names are empty (because before there were always at least the internal dns names)

This was a bug. It doesn't make sense to add internal dns names for external listener because the DNS names should match the DNS names of the listener... Which now makes me think that we should not allow external+TLS+empty subdomain 🤔 , or else, what should we put in the DNS names of the cert?

WDYT?

Reply:
I've added an error in the case there's a listener with external+TLS+empty subdomain